### DISCUSSION TITLE
ePub incorrect rendering

### DISCUSSION THREAD

**Discussion by nata1993** [sumatrapdfreader/sumatrapdf#2835] [Q&A]
There looks like to be a rendering problem with some symbols as I have found out on forum with fractions. But there is another symbol that gets rendered not well. The symbol is "—". In ePub files before this symbol there is added a square for some reason even though between this symbol and previous symbol there is no blank space. The symbol may be put inside a single word yet the square is being still rendered right before this symbol. 
I have seen this problem with ePub, kepub and advanced-epub but have not yet seen this problem with Amazon Kindle file format AZW3.
The exact same square can be seen in this thread about fractions: https://github.com/sumatrapdfreader/sumatrapdf/discussions/2794
The difference between fractions not being rendered and — symbol rendering problem is that with fractions there are no fractions and only a square but with — the symbol gets rendered right next to the square.

Please advise, may it be rendering problem or may the problem be introduced when the epub files were initially created?
So far the problem occurs on the latest SumatraPDF version and have not yet tried other versions prior epub rendering engine update.

Edit: Found other epub file where supposedly same — symbol has been used yet there are no squared immediately before such symbol.

**Comment by halikular** [comment]
I haven't used sumatra before, but I suspect it has to do with the new rendering engine for epub. Spacing, alignment, and fonts, pretty much everything looks different from what  I would assume it should look like. The rendering in thorium and play books web looks like what I expect. I'll share screenshots to highlight some of the differences. I'm using sumatra version 3.4.6 and play books reading standard ebooks compatible epub release of [Don Quixote, by Miguel de Cervantes](https://standardebooks.org/ebooks/miguel-de-cervantes-saavedra/don-quixote/john-ormsby).
![image](https://user-images.githubusercontent.com/16994036/189392683-6f07c2d0-8f62-4d77-95b9-f9efa62ee81e.png)
![image](https://user-images.githubusercontent.com/16994036/189392767-7ddd8240-532c-461e-a917-5a382b907c78.png)
![image](https://user-images.githubusercontent.com/16994036/189392871-41487f58-a083-4f97-b63f-035baa7ed884.png)
![image](https://user-images.githubusercontent.com/16994036/189392898-acf42c3b-6665-4a40-b8ec-a6ce1eddeeff.png)

**Comment by GitHubRulesOK** [comment]
the missing symbol(s) includes possibly `%E2%81%A0` which is probably a form of `''`
